src/app/app.component.ts

```diff
@@ -1,19 +1,24 @@
+/* eslint-disable @ngrx/prefer-selector-in-select */
 import { Component, OnInit } from '@angular/core';
 import { Store } from '@ngrx/store';
-import { RecipesService } from './services/recipes-service';
-import { updateRecipes } from './store/recipes/recipes.actions';
+import { RecipesService } from './recipes/services/recipes-service';
+import { RecipesActions } from './store/recipes';
+import { AppState } from './store/store';
 
 @Component({
 	selector: 'app-root',
 	templateUrl: './app.component.html',
 	styleUrls: ['./app.component.scss'],
 })
 export class AppComponent implements OnInit {
-	constructor(private recipesService: RecipesService, private store: Store) {}
+	constructor(
+		private recipesService: RecipesService,
```

Comment:
move business logic away from app.component, use app.component only for init logic (tabs titles logic, unsupported browsers etc). create `dashboard.component` or something similar and move it there

Reply:
I move recipes downloading to router resolver. Resolver dispatch action and effect responsible for get jokes.